src/hooks/Matches.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { JSX, useEffect, useState } from 'react';
+import {
+  Box,
+  Typography,
+  Card,
+  CardContent,
+  CardHeader,
+  Avatar,
+  CircularProgress,
+  Alert,
+  Stack,
+} from '@mui/material';
+import { Match } from '../types/MatchesType';
+import { UpcomingMatchesProps } from '../constants/Game';
+import styles from '../hooks/Styles.module.scss';
+
+const UpcomingMatches: React.FC<UpcomingMatchesProps> = ({ game }): JSX.Element => {
+  const [matches, setMatches] = useState<Match[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+
+  const API_TOKEN = process.env.REACT_APP_PANDASCORE_TOKEN!;
+  const BASE_URL = 'https://api.pandascore.co';
+
+  useEffect(() => {
+    const fetchMatches = async () => {
```

Comment:
please read issue description, this approach is not what i was asking about, you need to create api service and custom hooks